File: 3.EDD/4°Version/Java/Financiera.java
import java.util.Scanner;

public class Financiera {
    
    public static void recorridoNormal(nodoPrestamo iterador){
        if(iterador != null){
            System.out.println(iterador.getPrestamo().ToString());
            recorridoNormal(iterador.getEnlace());
        }
    }

    public static void recorridoInvertido(nodoPrestamo iterador){
        if(iterador != null){
            recorridoInvertido(iterador.getEnlace());
            System.out.println(iterador.getPrestamo().ToString());
        }
    }
    
    public static void main (String[] args){
        Scanner mex = new Scanner (System.in);
        nodoPrestamo inicio = new nodoPrestamo();

        int continuar =1;

        while(continuar != 0){
            Prestamo P1 = new Prestamo();

            System.out.println("\n \n");
            System.out.println("Datos del acreedor");
            System.out.println("\n");
            System.out.println("Ingresa los nombres: ");
            P1.setNombre(mex.nextLine());
            System.out.println("Ingresa los apellidos: ");
            P1.setApellido(mex.nextLine());
            System.out.println("Ingresa un nuemro telefonico: ");
            P1.setTelefono(mex.nextLine());
            System.out.println("Ingresa un correo electronico (Opcional): ");
            P1.setCorreo(mex.nextLine());

            System.out.println("Ingrese la cantidad a prestar: ");
            P1.setPrestamo(mex.nextDouble());
            System.out.println("Ingresar el porcentaje de interes: ");
            P1.setPorcentaje(mex.nextDouble());
            System.out.println("Ingresa la cantidad de a pagar: ");
            P1.setSemana(mex.nextDouble());
            System.out.println("El prestamo a sido registrado");

            nodoPrestamo nuevo  = new nodoPrestamo(P1);
            nodoPrestamo iterador;
            iterador = inicio;

            while (iterador.getEnlace() != null) {
                iterador = iterador.getEnlace();
                
            }
            iterador.setEnlace(nuevo);

            System.out.println("Para continuar oprima 1, para finalizar 0: ");
            continuar = mex.nextInt();
            mex.nextLine();
        }

        System.out.println("\n \n");
        System.out.println("------------------------------------------------------------");
        System.out.println("\n Lista Normal \n");
        recorridoNormal(inicio.getEnlace());

        System.out.println("\n \n");
        System.out.println("------------------------------------------------------------");
        System.out.println("\n Lista Invertida \n");
        recorridoInvertido(inicio.getEnlace());
        mex.close();
    }
}

File: 3.EDD/4°Version/C++/Financiera.cpp
#include <iostream>
#include "nodoPrestamo.h"
using namespace std;

void recorrerNormal(nodoPrestamo *iterador){
    if(iterador){
        cout << "\n \n";
        cout << iterador->getPresta().toString();

        recorrerNormal(iterador->getEnlace());
    }
}

void recorrerInvertida(nodoPrestamo *iterador){
    if(iterador){
        recorrerInvertida(iterador->getEnlace());
        cout << "\n \n";
        cout << iterador->getPresta().toString();
    }
}

int main(int argc, char *argv[]){

    nodoPrestamo *inicio = new nodoPrestamo();
    int continuar;

    string palabras;
    double cantidad;
    Prestamo P1;

    do{
        // Datos del acreedor-------------------
        cout << "Ingresar datos del acreedor";
        cout << "\n \n";
        cout << "Ingresar nombres: ";
        getline(cin,palabras);
        P1.setNombre(palabras);
        cout << "Ingresar apellidos: ";
        getline(cin,palabras);
        P1.setApellido(palabras);
        cout << "Ingresar un numero telefonico: ";
        getline(cin,palabras);
        P1.setTelefono(palabras);
        cout << "Ingresar correo electronico (Opcional): ";
        getline(cin,palabras);
        P1.setCorreo(palabras);

        //Datos del pretamo-------------------------
        cout <<"\n \n";
        cout << "Ingresar la cantidad a prestar: ";
        cin >> cantidad;
        P1.setPrestamo(cantidad);
        cout << "Ingresar el porcentaje de interes (%): ";
        cin >> cantidad;
        P1.setPorcentaje(cantidad);
        cout << "Ingresar la cantidad de semanas a pagar: ";
        cin >> cantidad;
        P1.setSemana(cantidad);

        nodoPrestamo *nodoP1= new nodoPrestamo(P1);
       
        nodoPrestamo*iterador= inicio;
        while( iterador->getEnlace() != NULL ){ 
            iterador= iterador->getEnlace();
        }
        iterador->setEnlace( nodoP1 );   
        cout << "... .. . Prestamo registrado!\n\n";
       
        cout << "Presiona 1 Para continuar y o para finalizar: ";
        cin >> continuar;
        cin.ignore();

    }while(continuar == 1);

    cout << "\n \n";
    cout << "--------------------------------------------------------------------------------------";
    cout << "\n Recorrido de lista Normal \n";
    recorrerNormal(inicio->getEnlace());

    cout << "\n \n";
    cout << "--------------------------------------------------------------------------------------";
    cout << "\n Recorrido de lista invertidaad \n";
    recorrerInvertida(inicio->getEnlace());

    return 0;
}